pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -20,7 +20,10 @@
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
 , enableSharedLibraries ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
-, enableSplitObjs ? !stdenv.isDarwin # http://hackage.haskell.org/trac/ghc/ticket/4013
+# For GHC 8.0.2 we use split-sections instead
+# Unsupported for Darwin/Clang, see http://hackage.haskell.org/trac/ghc/ticket/4013
+, enableSplitObjs ? (!stdenv.isDarwin && (stdenv.lib.versionAtLeast "8.0.1" ghc.version))
+, enableSplitSections ? (!stdenv.isDarwin && (stdenv.lib.versionOlder "8.0.1" ghc.version))
```

Comment:
Do we really need a new argument `enableSplitSections` for this change? It seems to me like "split objects" and "split sections" are two different implementations of the same concept, and IMHO it seems fine to control both with *one* option.